ibis/expr/types/temporal.py

```diff
@@ -19,9 +48,215 @@ class TemporalColumn(AnyColumn, TemporalValue):
     pass  # noqa: E701,E302
 
 
+class DateComponentValue(TemporalValue):
+    """Temporal expressions that have a date component.
+
+    Currently this includes
+    [`TimestampValue`][ibis.expr.types.temporal.TimestampValue]s and
```

Comment:
Does this conform with numpydoc rules?

Reply:
It conforms with mkdocs and renders correctly.

Reply:
I don't think numpydoc says anything about how objects are referenced, it only specifies docstring section meanings AFAIK.

Reply:
Apparently `pydocstyle` doesn't like our `f`-string docs. Otherwise we could use that. `numpydoc` is too primitive: it validates one module at a time.